### ISSUE TITLE
multiple rules for same domain in ingressroute ignores given tls options

### ISSUE THREAD

**Issue by pflipp** [traefik/traefik#7385]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?
* created a TLSOptions Resource
* created a Certificate which is valid for two domains host1.example.com & host2.example.com
* created a IngressRoute which contains three rules for the two domains
* i reused a http2 connection established on host2.example.com for host1.example.com

### What did you expect to see?
Response from host2.example.com as the tls options are the same

### What did you see instead?
421 Misdirected Request as traefik falls back to default tls options  

when removing the rule with the PathPrefix it works as expected

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.3.1
Codename:     picodon
Go version:   go1.15.2
Built:        2020-09-29T15:49:06Z
OS/Arch:      linux/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?
Provider: Kubernetes

```yaml
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRoute
metadata:
  name: example-route
  namespace: example
spec:
  entryPoints:
    - secure
  routes:
    - kind: Rule
      match: Host(`host1.example.com`)
      middlewares:
        - name: headers-ssl
          namespace: traefik2
        - name: headers
          namespace: traefik2
        - name: host1-feature-policy
          namespace: example
        - name: browser-security
          namespace: traefik2
      services:
        - kind: Service
          name: service1
          namespace: example
          port: 3000
    - kind: Rule
      match: Host(`host2.example.com`) && PathPrefix(`/some-path`)
      middlewares:
        - name: headers-ssl
          namespace: traefik2
        - name: headers
          namespace: traefik2
        - name: strip-prefix-example
          namespace: example
      services:
        - kind: Service
          name: service2
          namespace: example
          port: 4445
    - kind: Rule
      match: Host(`host2.example.com`)
      middlewares:
        - name: headers-ssl
          namespace: traefik2
        - name: headers
          namespace: traefik2
      services:
        - kind: Service
          name: service2
          namespace: example
          port: 4444
  tls:
    options:
      name: example-tls-option
      namespace: traefik2
    secretName: example-cert

```

I may also provide more resources but I think looking at https://github.com/traefik/traefik/blob/f2e53a356946454e7fc366fce35378973ce43fd1/pkg/server/router/tcp/router.go#L154 makes clear where the problem is.
Maybe it also should check if the tls options are diffrent?

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
[traefik2-ingress-controller-r84l6] time="2020-10-06T09:38:01Z" level=debug msg="TLS options difference: SNI=default, Header:traefik2-example-tls-option" host=host1.example.com req.Host=host1.example.com req.TLS.ServerName=host2.example.com
```

**Comment by jspdown** [comment]
Hello @pflipp and thanks for reporting this bug. We have been able to reproduce it consistently and you are right on the source of the bug.

**Comment by pflipp** [comment]
nice @jspdown
thank you for investigating and providing the PR!

**Comment by traefiker** [comment]
Closed by #7609.
